proxy_collection.go

```diff
@@ -61,6 +66,49 @@ func (collection *ProxyCollection) AddOrReplace(proxy *Proxy, start bool) error
 	return nil
 }
 
+func (collection *ProxyCollection) PopulateJson(data io.Reader) ([]*Proxy, error) {
```

Comment:
I don't think this should care about the `io.Reader`, it should probably just get the input struct directly regardless of serialization format.